python/cudf/dataframe/index.py

```diff
@@ -702,6 +712,9 @@ def as_index(arbitrary, name=None):
         - GenericIndex for all other inputs.
     """
     # This function should probably be moved to Index.__new__
+    if hasattr(arbitrary, 'name') and name is None:
+        name = arbitrary.name
+
     if isinstance(arbitrary, Index):
         return arbitrary
```

Comment:
You may want to pass the name here as well, if arbitrary is unnamed and a user passes the name as a parameter you'll end up unnamed.

Reply:
Yep handling it now. Thanks for catching this slipped.